src/plugins/intel_gpu/tests/test_cases/gemm_gpu_test.cpp

```diff
@@ -342,6 +342,178 @@ TEST(gemm_gpu, dynamic) {
     }
 }
 
+TEST(gemm_gpu, dynamic_multi_inference_same_shape) {
+    auto& engine = get_test_engine();
+
+    auto in1_dyn_layout = layout{ ov::PartialShape{ 1, 1, ov::Dimension(1, 10), 4 }, data_types::f32, format::bfyx };
+    auto in1_actual_layout = layout{ ov::PartialShape{ 1, 1, 3, 4 }, data_types::f32, format::bfyx };
+    auto in2_layout = layout{ ov::PartialShape{ 4, 1 }, data_types::f32, format::bfyx };
```

Comment:
it's better to use dynamic shape for second input as well to ensure that it's handled properly in the kernel

Reply:
done